Engine/DataFeeds/SubscriptionUtils.cs

```diff
@@ -97,7 +97,8 @@ public static class SubscriptionUtils
                         var mode = requestMode != DataNormalizationMode.Raw
                             ? requestMode
                             : DataNormalizationMode.Adjusted;
-                        if (enablePriceScale && data?.Time.Date > lastTradableDate)
+                        // We update our price scale factor when the date changes for non fill forward bars or if we haven't initialized yet
+                        if (enablePriceScale && data?.Time.Date > lastTradableDate && (!data.IsFillForward || lastTradableDate == DateTime.MinValue))
```

Comment:
Please add a regression test for:
 -  illiquid security, 
 - fill forwarding overnight
 - split overnight
 - what is the price at 931am the following morning? 
=========
I'd expect with this change -- it won't adjust the scale factor, it'll fill at yesterday's close -> which will result in an immediate jump in equity, once the first split-adjusted price arrives. 
=========

Reply:
A regression test reproducing the issue has been added, thanks @AdalyatNazirov.
It's placing a trade on pre-market where fill forward data is used, one thing to note is that it won't fill using fill forward data but wait till the first data point comes in. The same will happen with all order types because it's using the assets last bar time **date** and comparing it with the assets time https://github.com/QuantConnect/Lean/blob/master/Common/Orders/Fills/FillModel.cs#L547 which will be false until a new non fill forward, data point comes in

Reply:
Something to note, is that another issue uncovered by the regression algorithm (and covered) is that auxiliary data points shouldn't update the price scale either